src/main/java/duke/commands/DeleteCommand.java

```diff
@@ -0,0 +1,74 @@
+package duke.commands;
+
+import java.io.IOException;
+
+import duke.exceptions.DukeException;
+import duke.exceptions.TaskNotFoundException;
+import duke.exceptions.TaskSelectionInvalidException;
+import duke.storage.Storage;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Responsible for dealing with the deletion of tasks.
+ */
+public class DeleteCommand extends Command {
+    private String[] checkCommands;
+
+    /**
+     * Constructs a DeleteCommand with the given full command line.
+     *
+     * @param fullCommand Full command line input.
+     */
+    public DeleteCommand(String fullCommand) {
+        this.checkCommands = fullCommand.split(" ");
```

Comment:
Is the `this` qualifier necessary here? I noticed this in several other files as well.